assertion failed in WriteBits

Hey, not sure if this is an issue in google/brotli, this, or the way I've used it but I was wondering if you could give a look at https://github.com/itchio/butler/issues/3

You can browse the v0.4.5 tree at https://github.com/itchio/butler/tree/v0.4.5 — the `testBrotli` function lives at https://github.com/itchio/butler/blob/v0.4.5/butler.go#L151

Basically you should just be able to feed `butler.exe` (the v0.4.5 windows-amd64 build, which you can grab from [this .7z archive](https://misc.amos.me/butler/windows-amd64/v0.4.5/butler.7z)) to a compress/decompress test & be able to reproduce.

I'll try to come up with a stack trace & a better/more self-contained test case later on, but if you've already encountered that / have any idea where it might come from, I'd be glad to know about it. Thanks!

```
amos at bowser in ~/tmp/brotli (master)
$ go install gopkg.in/kothar/brotli-go.v0/gbr
# gopkg.in/kothar/brotli-go.v0/gbr
C:\Go\pkg\tool\windows_amd64\link.exe: running g++ failed: exit status 1
C:\msys64\tmp\go-link-991828467/000001.o:mingw_helpers.c:(.text+0x14340): multiple definition of `___chkstk_ms'
C:\msys64\tmp\go-link-991828467/000000.o:streams.c:(.text+0x6d30): first defined here
collect2.exe: error: ld returned 1 exit status
```

Those flags are an ugly workaround btw, go folks consider it a bug in cgo they want to resolve another way, cf. https://github.com/golang/go/issues/9510 and https://golang.org/cl/16741

I'm wondering if these flags could cause the WriteBits issue.. but I don't really see how. I'll keep digging. Currently trying to compile `google/brotli` to see if the go wrapper is involved in the issue at all.

--- a/enc/write_bits.h
+++ b/enc/write_bits.h
@@ -49,7 +49,7 @@ inline void WriteBits(int n_bits,
 #ifdef BIT_WRITER_DEBUG
   printf("WriteBits  %2d  0x%016llx  %10d\n", n_bits, bits, *pos);
 #endif
-  assert(bits < 1UL << n_bits);
+  assert((bits >> n_bits) == 0);
 #ifdef IS_LITTLE_ENDIAN
   // This branch of the code can write up to 56 bits at a time,
   // 7 bits are lost by being perhaps already in *p and at least

